File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/ChampionImageDao.java
package com.legue.axel.lolsummonertool.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.ChampionImage;

import java.util.List;

@Dao
public interface ChampionImageDao {

    @Query("SELECT * FROM images ORDER BY id")
    LiveData<List<ChampionImage>> getChampionImages();

    @Query("SELECT * FROM images WHERE id = :riotImageId")
    LiveData<ChampionImage> getChampionImageById(int riotImageId);

    @Query("SELECT * FROM images WHERE championId = :championKey")
    LiveData<ChampionImage> getChampionImageByChampionId(int championKey);

    @Query("SELECT * FROM images WHERE championId = :championKey")
    ChampionImage getChampionImageByChampionIdWidget(int championKey);

    @Insert
    void insertChampionImage(ChampionImage championImage);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllChampionImage(List<ChampionImage> championImageList);

    @Delete
    void deleteChampionImage(ChampionImage championImage);

    @Query("DELETE FROM images")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateChampionImage(ChampionImage championImage);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/ChampionImageDao.kt
package com.legue.axel.lolsummonertool.database.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.ChampionImage

@Dao
interface ChampionImageDao {

    @Query("SELECT * FROM images ORDER BY id")
    fun getChampionImages(): LiveData<List<ChampionImage>>

    @Query("SELECT * FROM images WHERE id = :riotImageId")
    fun getChampionImageById(riotImageId: Int): LiveData<ChampionImage>

    @Query("SELECT * FROM images WHERE championId = :championKey")
    fun getChampionImageByChampionId(championKey: Int): LiveData<ChampionImage>

    @Query("SELECT * FROM images WHERE championId = :championKey")
    fun getChampionImageByChampionIdWidget(championKey: Int): ChampionImage

    @Insert
    fun insertChampionImage(championImage: ChampionImage)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllChampionImage(championImageList: List<ChampionImage>)

    @Delete
    fun deleteChampionImage(championImage: ChampionImage)

    @Query("DELETE FROM images")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateChampionImage(championImage: ChampionImage)
}
